addons/product/product_view.xml

```diff
@@ -53,13 +53,14 @@
             <field name="model">product.template</field>
             <field name="arch" type="xml">
                 <tree string="Product">
+                    <field name="default_code"/>
```

Comment:
The handle widget is always in the first position.
Invert default_code and handle widget.